### PR TITLE
Change "Metrics" to "Infrastructure"

### LINKED ISSUE #2097: [Request] Change "Metrics" to "Infrastructure"
## Description

It looks like it was decided in https://github.com/elastic/kibana/issues/123972 that the high-level "Metrics" section of the Observability app would be changed to "Infrastructure", and the changes were made for 8.4 in https://github.com/elastic/kibana/pull/135278. This affects the text, screenshots and navigation elements (titles, table of contents, etc) throughout the Observability docs.

👋  @jasonrhodes @matschaffer (just pulling from https://github.com/elastic/kibana/issues/123972 and https://github.com/elastic/kibana/pull/135278, but feel free to tag others) what can we do to make sure these changes are communicated to the docs team earlier in the process? We happened to find out about this change in an [unrelated PR review](https://github.com/elastic/observability-docs/pull/2089#pullrequestreview-1073957096). 

## Collaboration

The docs team (@colleenmcginnis) will lead producing the content

**Contact Person:** @jasonrhodes @matschaffer can one of you serve as contact person and reviewer for this work? Or can you tag someone else in?

## Suggested Target Release

8.4

## Stakeholders

Anyone else?

### PR DESCRIPTION
**Pull request by colleenmcginnis**
Closes #2097

## Overview

Changes references to the Metrics app and relevant UI components to use "Infrastructure" instead of "Metrics".

Replaces:
- "Metrics app" → "Infrastructure app"
- "Metrics" in Kibana's main menu → "Infrastructure"
- "Metrics" in Observability menu → "Infrastructure"
- "Metrics" in breadcrumb → "Infrastructure"
- Some instances of "metrics" where I think it would be helpful to hint at what part of the UI it's referring to (this is subjective and I'm open to feedback!)

Does **not** replace:
- "Metrics threshold rule" in "Alerts and rules"
- "Metrics Explorer" as a page in the Infrastructure app
- "Metrics indices" on the Infrastructure Settings page
- "Metrics System" in the title of the System integration's dashboard
- "Metrics" tab in Host details
- Any use of "metrics" in filenames, section IDs, or page IDs (URLs)

## For the reviewer

- Does the Infrastructure app have "Explore data"? We [documented it for the Metrics app](https://www.elastic.co/guide/en/observability/current/exploratory-data-visualizations.html#create-multi-series-visualizations), but it doesn't look like it exists there anymore.
- Note: There were very few screenshots with "Metrics" in them since most were cropped to exclude the table of contents on the left side and the breadcrumbs along the top.

## To do

- [x] @colleenmcginnis Update text:
    - [x] Observability guide (https://github.com/elastic/observability-docs/commit/c071b07cf29f1a95c6d274e8cf9abc772cde51c1)
    - [x] ~~Integrations guide~~
    - [x] ~~Fleet guide~~
- [x] @colleenmcginnis Update screenshots:
    - [x] Observability guide
    - [x] ~~Integrations guide~~
    - [x] ~~Fleet guide~~
- [x] Review by:
    - [x] Product team (@jasonrhodes @matschaffer? @neptunian )
    - [x] Docs team (@elastic/obs-docs)
- [x] @colleenmcginnis address feedback
- [x] @colleenmcginnis merge /docs PR with new shared attribute (https://github.com/elastic/docs/pull/2501)
    - [x] Remove temporary book-scoped attribute in this repo (https://github.com/elastic/observability-docs/pull/2102/commits/774c56c443628a330dfbef5ce28936faabc21693)
- [ ] @colleenmcginnis merge this PR after ☝️ is merged

### CODE REVIEW

**Review by dedemorton — APPROVED** [review]
LGTM! I noticed a couple of inconsistencies with how we do menu paths, but we should fix that later with some linting because we have that problem in many places. :-) Thanks for doing this!

**Comment by jasonrhodes** [comment]
@colleenmcginnis :

> Does the Infrastructure app have "Explore data"? We [documented it for the Metrics app](https://www.elastic.co/guide/en/observability/current/exploratory-data-visualizations.html#create-multi-series-visualizations), but it doesn't look like it exists there anymore.

No, it was removed. https://github.com/elastic/kibana/pull/125109#issuecomment-1076773371

File: docs/en/observability/view-observability-alerts.asciidoc [hunk 1]
```diff
@@ -10,7 +10,7 @@ For more information on configuring alerts, see <<create-alerts-configure>>.
 ****
 
 The Alerts page lists all the alerts that have met a condition defined by a rule you created using the Logs,
-Metrics, Uptime, or APM apps.
+Infrastructure, Uptime, or APM apps.
```
Comment (jasonrhodes) [inline docs/en/observability/view-observability-alerts.asciidoc:13]: This is good but I do wonder if we shouldn't save ourselves future hassle and say "created using one of the Observability apps" in places like this?

File: docs/en/observability/view-monitor-status.asciidoc [hunk 2]
```diff
@@ -61,7 +61,7 @@ image::images/monitors-list.png[Monitors list]
 
 [discrete]
 [[observability-integrations]]
-== Integrate with Logs, Metrics, and APM
+== Integrate with Logs, Infrastructure, and APM
```
Comment (jasonrhodes) [inline docs/en/observability/view-monitor-status.asciidoc:64]: Does it make sense to move to something more like "Integrate with Observability" or "Integrate with Other Observability Apps" etc?

**Comment by colleenmcginnis** [comment]
@jasonrhodes @matschaffer @neptunian I will plan to merge this before 9am PDT tomorrow if there are no other comments/concerns posted here before then.

**Comment by jasonrhodes** [comment]
@colleenmcginnis thanks sounds good. I left a few comments earlier today but nothing major for now.

**Review by matschaffer — APPROVED** [review]
Looks good! Thanks and sorry again for for not including this in my changes 🧡

**Review by neptunian — APPROVED** [review]

**Pull request merged**